internal/server/statvar_search.go

```diff
@@ -0,0 +1,85 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package server
+
+import (
+	"context"
+	"sort"
+	"strings"
+
+	pb "github.com/datacommonsorg/mixer/internal/proto"
+)
+
+// SearchStatVar implements API for Mixer.SearchStatVar.
+func (s *Server) SearchStatVar(
+	ctx context.Context, in *pb.SearchStatVarRequest) (
+	*pb.SearchStatVarResponse, error) {
+	query := in.GetQuery()
+	result := &pb.SearchStatVarResponse{
+		StatVars:      []string{},
+		StatVarGroups: []string{},
+	}
+	if query == "" {
+		return result, nil
+	}
+	tokens := strings.Split(
+		strings.Replace(strings.ToLower(query), ",", " ", -1), " ")
```

Comment:
We don't want to only split by a single space because the token string could be separated by more than a single space or have a space after a comma. eg. "token1  token2" or "token1, token2" which would also become "token1  token2"

I think strings.Fields does what we would want.

Might also want to add tests for these cases of multiple white spaces

Reply:
Updated and added tests.